Source/monster.cpp

```diff
@@ -2846,29 +2843,34 @@ bool M_DumbWalk(int i, Direction md)
 
 bool M_RoundWalk(int i, Direction md, int *dir)
 {
-	if (*dir)
-		md = left[left[md]];
+	if (*dir != 0)
+		md = left[left[md]]; // 90 degrees left
 	else
-		md = right[right[md]];
+		md = right[right[md]]; // 90 degrees right
+
+	Direction turn90deg = md;
 
-	Direction mdtemp = md;
 	bool ok = DirOK(i, md);
-	if (!ok) {
-		if (*dir) {
-			md = right[mdtemp];
-			ok = DirOK(i, md) || (md = right[right[mdtemp]], DirOK(i, md));
+
+	for (int j = 0; !ok && j < 2; j++) {
+		// Keep trying up to an including backwards

```

Comment:
This looks to me like it's turning the opposite direction from the original 90-degree turn. So it's not turning the monster backwards, but rather forwards.

```suggestion
		// Keep trying by turning back toward the original direction
```

Reply:
Oh you are right, I might rethink how this is done in that case.